pkg/controllers/machine_controller.go

```diff
@@ -290,65 +298,133 @@ func (r *MachineReconciler) deleteMachine(ctx context.Context, log logr.Logger,
 
 	log.V(1).Info("Finalizer present, doing cleanup")
 
-	log.V(1).Info("Removing volumes")
-	if err := r.deleteVolumes(ctx, log, machine); err != nil {
-		return fmt.Errorf("error removing machine disks: %w", err)
-	}
-	log.V(1).Info("Successfully removed machine disks")
+	domain, err := r.libvirt.DomainLookupByUUID(libvirtutils.UUIDStringToBytes(machine.ID))
+	if err != nil {
 
-	log.V(1).Info("Removing network interfaces")
-	if err := r.deleteNetworkInterfaces(ctx, log, machine); err != nil {
-		return fmt.Errorf("error removing machine network interfaces: %w", err)
+		if libvirt.IsNotFound(err) {
+			err = r.cleanupAdditionalResources(ctx, log, machine)
+			if err != nil {
+				r.queue.AddAfter(machine.ID, defaultRetryReconcileTime)
+				return err
+			}
+			log.V(1).Info("Cleaned up all dependencies, removing finalizer")
+			machine.Finalizers = utils.DeleteSliceElement(machine.Finalizers, MachineFinalizer)
+			if _, err := r.machines.Update(ctx, machine); store.IgnoreErrNotFound(err) != nil {
+				return fmt.Errorf("failed to update machine metadata: %w", err)
```

Comment:
will it requeue again?

Reply:
yes, the control will go to line 198, and item will be requeued.
(https://github.com/ironcore-dev/libvirt-provider/blob/8c953a3bdf012e2ff2a4f6732027a21679b4e7cd/pkg/controllers/machine_controller.go#L198C3-L198C31)